src/backend/commands/explain_gp.c

```diff
@@ -2210,3 +2213,159 @@ void ExplainParallelRetrieveCursor(ExplainState *es, QueryDesc* queryDesc)
+						 "inlining_counter %.3f ms, optimization_counter %.3f ms, emission_counter %.3f ms,"
+						 "total_time %.3f ms.",
+						 slice_index,
+						 ss->segindex0 + j,
+						 ji->created_functions,
+						 1000.0 * INSTR_TIME_GET_DOUBLE(ji->generation_counter),
+						 1000.0 * INSTR_TIME_GET_DOUBLE(ji->inlining_counter),
+						 1000.0 * INSTR_TIME_GET_DOUBLE(ji->optimization_counter),
+						 1000.0 * INSTR_TIME_GET_DOUBLE(ji->emission_counter),
+						 1000.0 * INSTR_TIME_GET_DOUBLE(total_time));
+
+			avg_time += INSTR_TIME_GET_DOUBLE(total_time);
+			if (INSTR_TIME_GET_DOUBLE(total_time) > max_time)
+			{
+				max_time = INSTR_TIME_GET_DOUBLE(total_time);
+				idx2 = j;
+			}
+			nworker++;
+		}
+
+		// print nothing if jit is not performed on all workers in current slice
+		if (nworker == 0)
+			continue;
+
+		avg_functions /= nworker;
+		avg_time /= nworker;
+		if (es->format == EXPLAIN_FORMAT_TEXT)
+		{
+			appendStringInfoSpaces(es->str, es->indent * 2);
+			appendStringInfo(es->str, "(slice%d): ", slice_index);
+			appendStringInfo(es->str, "Functions: ");
+			if (ss->nworker == 1)
+				appendStringInfo(es->str, "%.2f. ", max_functions);
+			else
+				appendStringInfo(es->str, "%.2f avg x %d workers, %.2f max (seg%d). ",
+								 avg_functions, nworker, max_functions, ss->segindex0 + idx1);
+
+			if (es->analyze && es->timing)
+			{
+				appendStringInfo(es->str, "Timing: ");
+				if (ss->nworker == 1)
+					appendStringInfo(es->str, "%.3f ms total.\n", 1000.0 * max_time);
+				else
+					appendStringInfo(es->str, "%.3f ms avg x %d workers, %.3f ms max (seg%d).\n",
+									 1000.0 * avg_time, nworker, 1000.0 * max_time, ss->segindex0 + idx2);
+			}
+		}
+		else
+		{
+			ExplainOpenGroup("slice", "slice", true, es);
+			ExplainPropertyInteger("slice", NULL, slice_index, es);
+			if (ss->nworker == 1)
+				ExplainPropertyFloat("functions", NULL, max_functions, 2, es);
+			else
+			{
+				ExplainOpenGroup("Functions", "Functions", true, es);
+				ExplainPropertyFloat("avg", NULL, avg_functions, 2, es);
+				ExplainPropertyInteger("nworker", NULL, nworker, es);
+				ExplainPropertyFloat("max", NULL, max_functions, 2, es);
+				ExplainPropertyInteger("segid", NULL, ss->segindex0 + idx1, es);
+				ExplainCloseGroup("Functions", "Functions", true, es);
+			}
+
+			if (es->analyze && es->timing)
+			{
+				if (ss->nworker == 1)
+					ExplainPropertyFloat("Timing", NULL, max_time, 3, es);
+				else
+				{
+					ExplainOpenGroup("Timing", "Timing", true, es);
+					ExplainPropertyFloat("avg", NULL, 1000.0 * avg_time, 3, es);
+					ExplainPropertyInteger("nworker", NULL, nworker, es);
+					ExplainPropertyFloat("max", NULL, 1000.0 * max_time, 3, es);
+					ExplainPropertyInteger("segid", NULL, ss->segindex0 + idx2, es);
+					ExplainCloseGroup("Timing", "Timing", true, es);
+				}
+			}
+			ExplainCloseGroup("slice", "slice", true, es);
+		}
+	}
+
+	es->indent -= 1;
+	ExplainCloseGroup("JIT", "JIT", true, es);
+}
```

Comment:
Should we add empty line in the end of file here?

Reply:
I haven't touched this file, so I guess not.